### PR TITLE
include expiration in audit log lock event

### PR DESCRIPTION
**Pull request by stevenGravy**
If a expiration was set it wasn't included in the audit log for lock create events.  Currenlty it's always 

`  "expires": "0001-01-01T00:00:00Z",`

Now it will set the expires as in this case when it's `2m` away.  `0001-01-01T00:00:00Z` will show for never expires.

```json
{
  "cluster_name": "teleport.example.com",
  "code": "TLK00I",
  "ei": 0,
  "event": "lock.created",
  "expires": "2023-05-17T02:18:35.44638Z",
  "name": "56642c1f-441b-46ae-ab4c-de4872c891f2",
  "target": {
    "role": "auditor"
  },
  "time": "2023-05-17T02:16:35.451Z",
  "uid": "5343ca78-4984-49de-932c-a2a665d93cc6",
  "updated_by": "steven",
  "user": "steven"
}
```

Updated test to include a time comparison.

changelog: Lock expiration time included in Audit Log Lock Create events

### CODE REVIEW

File: lib/auth/access.go [hunk 1]
```diff
@@ -116,6 +122,7 @@ func (a *Server) UpsertLock(ctx context.Context, lock types.Lock) error {
 		UserMetadata: um,
 		ResourceMetadata: apievents.ResourceMetadata{
 			Name:      lock.GetName(),
+			Expires:   expiresTime,
```
[the remark this answers is not in the document]
Comment (stevenGravy) [reply lib/auth/access.go:125]: It does have the `omitempty` in the `ResourceMetadata` structure for expires. Is there something else we can change?

File: lib/auth/access_test.go [hunk 2]
```diff
@@ -77,6 +78,8 @@ func TestUpsertDeleteLockEventsEmitted(t *testing.T) {
 	lock, err := types.NewLock("test-lock", types.LockSpecV2{
 		Target: types.LockTarget{MFADevice: "mfa-device-id"},
 	})
+	futureTime := time.Now().UTC().Add(12 * time.Hour)
+	lock.SetLockExpiry(&futureTime)
```
Comment (zmb3) [inline lib/auth/access_test.go:82]: This moved the error check too far away from the error. I would move this after the assertion.

Comment (stevenGravy) [reply lib/auth/access_test.go:83]: @zmb3 please see update

File: lib/auth/access.go [hunk 3]
```diff
@@ -190,6 +191,11 @@ func (a *Server) UpsertLock(ctx context.Context, lock types.Lock) error {
 		return trace.Wrap(err)
 	}
 
+	var expiresTime time.Time
+	// leave as 0 if no lock expiration was set
+	if lock.LockExpiry() != nil {
+		expiresTime = lock.LockExpiry().UTC()
+	}
```
Comment (zmb3) [inline lib/auth/access.go:198]:
```suggestion
	if le := lock.LockExpiry(); le != nil {
		expiresTime = le.UTC()
	}
```

**Review by zmb3 — APPROVED** [review]
minor suggestion

**Review by mdwn — APPROVED** [review]

**Pull request merged**